### DISCUSSION TITLE
Moving PointValue point as mesh is displaced

### DISCUSSION THREAD

**Discussion by Vandenbg** [idaholab/moose#21649] [Q&A General] [answered]
Hello,

I am making a nano-indentation model which involves driving a pyramid into a material. I'd like to know the value of stress at the tip of the pyramid as it is pressed into the material puck. I was thinking I could use a parsed function to determine the point as it changes position and feed that to the [PointValue](https://mooseframework.inl.gov/source/postprocessors/PointValue.html) PostProcessor, I have not been successful. Is this the correct way to go about getting this value? Or is there a better option? 

Here is a small section of my input which I am currently tinkering with. This does not work, however, it displays the logic which I am trying to achieve.  

[Functions]
  [point_value]
    type = ParsedFunction
    value = '0-t'
  []
[]
[PostProcessors]
  [strss_zz_tip]
    type = PointValue
    variable = stress_zz
    point = ' 0 point_value 12.5'
    block = 1
  []
[]

Thank you.

**Comment by GiudGiud** [comment, marked as answer]
Hello

No you may not specify a point coordinate using a function. 
This object expects 3 real numbers, not floating points. 

if you want to use PointValue you ll have to modify it to accept a function for each coordinate. This should not be very difficult at all. 

The PointValue execute already searches for the point on every execution (and not just on initialization) so you wont need to change much there. 

Guillaume

**Reply by Vandenbg** [reply]
Thank you for the reply,

I'll look into alternative solutions as I am hoping to accomplish this, or something similar, without editing the executables.

**Comment by maxnezdyur** [comment]
You might be able to use NodalVariableValue. You need to know the node id of the tip of the pyramid.

**Reply by Vandenbg** [reply]
This works well, thank you for pointing it out.
